src/main/java/community/whatever/onembackendjava/service/UrlShortenService.java

```diff
@@ -0,0 +1,37 @@
+package community.whatever.onembackendjava.service;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import org.springframework.util.StringUtils;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Random;
+
+@Slf4j
+@Service
+public class UrlShortenService {
+
+    private final Map<String, String> shortenUrls = new HashMap<>();
+
+    public String shortenUrlSearch(String key){
+        if (!shortenUrls.containsKey(key)) {
+            throw new IllegalArgumentException("Invalid key");
+        }
+        return shortenUrls.get(key);
+    }
+
+    public String shortenUrlCreate(String originUrl){
+        if (!StringUtils.hasText(originUrl)) {
+            throw new IllegalArgumentException("Request URL No exists ");
+        }
+        String randomKey;
+        while(true){
+            randomKey = String.valueOf(new Random().nextInt(10000000));
+            if (shortenUrls.putIfAbsent(randomKey, originUrl ) == null){
+                break;
+            }
+        }
+        return randomKey;
```

Comment:
```suggestion
        Random random = new Random();
        return Stream.generate(() -> String.valueOf(random.nextInt(10000000)))
                .filter(key -> shortenUrls.putIfAbsent(key, originUrl) == null)
                .findFirst()
                .orElseThrow();
```

`nit` 가독성만 수정한 코드 입니다. 적용할 것이라면 테스트가 필요 합니다.

Reply:
UrlShorten 서비스를 이용할때 응답 대기는 어느 정도의 시간까지 허용한다는 기준이 있을까요?